Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp

```diff
@@ -888,6 +892,66 @@ namespace AZ::Render
         debugDisplay->SetState(oldState);
     }
 
+    void AtomActorDebugDraw::UpdateActorInstance(EMotionFX::ActorInstance* actorInstance, float deltaTime)
+    {
+        // Find the corresponding trajectory trace path for the given actor instance
+        TrajectoryTracePath* trajectoryPath = FindTrajectoryPath(actorInstance);
+        if (!trajectoryPath)
+        {
+            return;
+        }
+
+        const EMotionFX::Actor* actor = actorInstance->GetActor();
+        const EMotionFX::Node* motionExtractionNode = actor->GetMotionExtractionNode();
+        if (motionExtractionNode)
+        {
+            const EMotionFX::Transform& worldTM = actorInstance->GetWorldSpaceTransform();
+
+            bool distanceTraveledEnough = false;
+            if (trajectoryPath->m_traceParticles.empty())
+            {
+                distanceTraveledEnough = true;
+            }
+            else
+            {
+                const size_t numParticles = trajectoryPath->m_traceParticles.size();
+                const EMotionFX::Transform& oldWorldTM = trajectoryPath->m_traceParticles[numParticles - 1].m_worldTm;
+
+                const AZ::Vector3& oldPos = oldWorldTM.m_position;
+                const AZ::Quaternion oldRot = oldWorldTM.m_rotation.GetNormalized();
+                const AZ::Quaternion rotation = worldTM.m_rotation.GetNormalized();
+
+                const AZ::Vector3 deltaPos = worldTM.m_position - oldPos;
+                const float deltaRot = AZStd::abs(rotation.Dot(oldRot));
+                if (deltaPos.GetLengthEstimate() > 0.0001f || deltaRot < 0.99f)
+                {
+                    distanceTraveledEnough = true;
+                }
+            }
+
+            // Add the time delta to the time passed since the last add
+            trajectoryPath->m_timePassed += deltaTime;
+
+            const uint32 particleSampleRate = 30;
+            if (trajectoryPath->m_timePassed >= (1.0f / particleSampleRate) && distanceTraveledEnough)
+            {
+                // Create the particle, fill its data and add it to the trajectory trace path
+                TrajectoryPathParticle trajectoryParticle;
+                trajectoryParticle.m_worldTm = worldTM;
+                trajectoryPath->m_traceParticles.emplace_back(trajectoryParticle);
+
+                // Reset the time passed as we just added a new particle
+                trajectoryPath->m_timePassed = 0.0f;
+            }
+        }
+
+        // Make sure we don't have too many items in our array
+        if (trajectoryPath->m_traceParticles.size() > 50)
+        {
+            trajectoryPath->m_traceParticles.erase(begin(trajectoryPath->m_traceParticles));
+        }
```

Comment:
Can you turn the 50 into a setting and also make sure to bring the array down to that number again? So erase(begin(), begin() + maxNumParticles) or something.

Reply:
will do!